### PR TITLE
Feature/form Add Login Page, Form Autofill, and UI Enhancements

### PR DESCRIPTION
**Pull request by santuch**
To test this update using this in terminal and open it in your localhost port 3000('http://localhost:3000/login.html')
npm install
node servertest.js

### Package and Scripts Updates:
* [`frontend/package.json`](diffhunk://#diff-da6498268e99511d9ba0df3c13e439d10556a812881c9d03955b2ef7c6c1c655L9-R15): Added `test` and `start` scripts, and added empty `author` and `description` fields.

### CSS File Renaming and Updates:
* [`frontend/public/css/form_styles.css`](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L18-R18): Renamed from `styles.css` and updated various styles including sidebar padding, heading sizes, container widths, button styles, and added hover effects for buttons. [[1]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L18-R18) [[2]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L30-R34) [[3]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L63-R65) [[4]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L73-R82) [[5]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4R101-R106) [[6]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L112-R120) [[7]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L127-R138) [[8]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L139-R154) [[9]](diffhunk://#diff-06969ab391eadd131881a753554b76343f1f34e6a6dd57b72511d4040b5b34c4L150-R170)

### New CSS File for Login Page:
* [`frontend/public/css/loginpage_styles.css`](diffhunk://#diff-4a9fed4df5799e946da8d233a4f16dc88abafd2af44dd9a4aaec8144339725e8R1-R173): Added a new CSS file with styles for the login page, including container, box, and button styles, as well as overlay and popup styles.

### HTML Form Updates:
* Updated form HTML files to reference the new `form_styles.css` and added `required` attributes to various input fields:
  * `frontend/public/form/delayed_registration.html` [[1]](diffhunk://#diff-14c4bde39f082e19974c272ed397e91db7d635a7ebee8d973cc5e6a156ff850eL10-R10) [[2]](diffhunk://#diff-14c4bde39f082e19974c272ed397e91db7d635a7ebee8d973cc5e6a156ff850eL21-R21) [[3]](diffhunk://#diff-14c4bde39f082e19974c272ed397e91db7d635a7ebee8d973cc5e6a156ff850eR230-R246) [[4]](diffhunk://#diff-14c4bde39f082e19974c272ed397e91db7d635a7ebee8d973cc5e6a156ff850eL268-R271)
  * `frontend/public/form/reg_request.html` [[1]](diffhunk://#diff-fdb65d94a38219e64bba16d8bf27047440fcb6545ec5cfac5253e4679a139129L11-R11) [[2]](diffhunk://#diff-fdb65d94a38219e64bba16d8bf27047440fcb6545ec5cfac5253e4679a139129L22-R22) [[3]](diffhunk://#diff-fdb65d94a38219e64bba16d8bf27047440fcb6545ec5cfac5253e4679a139129R233-R249) [[4]](diffhunk://#diff-fdb65d94a38219e64bba16d8bf27047440fcb6545ec5cfac5253e4679a139129L271-R274)
  * `frontend/public/form/resign.html` [[1]](diffhunk://#diff-0f54e811b67f689b35fa36bd4bff0f3b3fae6d6c7ac78285b9a46adaa75c24c1L11-R11) [[2]](diffhunk://#diff-0f54e811b67f689b35fa36bd4bff0f3b3fae6d6c7ac78285b9a46adaa75c24c1L22-R22) [[3]](diffhunk://#diff-0f54e811b67f689b35fa36bd4bff0f3b3fae6d6c7ac78285b9a46adaa75c24c1L203-R203) [[4]](diffhunk://#diff-0f54e811b67f689b35fa36bd4bff0f3b3fae6d6c7ac78285b9a46adaa75c24c1R218) [[5]](diffhunk://#diff-0f54e811b67f689b35fa36bd4bff0f3b3fae6d6c7ac78285b9a46adaa75c24c1R229) [[6]](diffhunk://#diff-0f54e811b67f689b35fa36bd4bff0f3b3fae6d6c7ac78285b9a46adaa75c24c1L239-R241)
  * `frontend/public/form/withdraw_course.html` [[1]](diffhunk://#diff-f673514e3b29116efd2bc42c4835ab6ef7f8b17b0d122e7bfdb85baa2fca3f68L11-R11) [[2]](diffhunk://#diff-f673514e3b29116efd2bc42c4835ab6ef7f8b17b0d122e7bfdb85baa2fca3f68L22-R22) [[3]](diffhunk://#diff-f673514e3b29116efd2bc42c4835ab6ef7f8b17b0d122e7bfdb85baa2fca3f68L124) [[4]](diffhunk://#diff-f673514e3b29116efd2bc42c4835ab6ef7f8b17b0d122e7bfdb85baa2fca3f68L137) [[5]](diffhunk://#diff-f673514e3b29116efd2bc42c4835ab6ef7f8b17b0d122e7bfdb85baa2fca3f68R229-R245) [[6]](diffhunk://#diff-f673514e3b29116efd2bc42c4835ab6ef7f8b17b0d122e7bfdb85baa2fca3f68L269-R270)update includes the addition of a login page, form autofill functionality, and various UI and UX improvements. Key changes:

Login Page: Added with styling and validation for a streamlined login experience.
Form Autofill: Implemented to automatically populate form fields with stored user data.
UI/UX Enhancements:

Validation & Popups: Added form validation, success pop-up, and hover effects on buttons.
Layout Improvements: Centered popup content, adjusted styling for better readability, and restructured sidebar for clarity.
Code Refinements:

Consolidated and cleaned up code for improved readability and maintainability.

Testing:
Validated login, autofill functionality, form submissions, and pop-up displays.

### CODE REVIEW

File: package-lock.json
Comment (krittamark) [inline package-lock.json:1]: This file don't need to exisst

File: package.json
Comment (krittamark) [inline package.json:1]: This file don't need to exist.

File: frontend/public/img/tu-logo.png
Comment (krittamark) [inline frontend/public/img/tu-logo.png:1]: The logo doesn't align in the center of the image.

**Review by krittamark — CHANGES_REQUESTED** [review]
The files `package-lock.json` and `package.json` in the root of the repository don't need to exist because frontend and backend run separately

**Comment by Santuch** [comment]
Remove files package-lock.json and package.json in the root and update logo file to tu-logo.webp

**Review by krittamark — APPROVED** [review]

**Pull request merged**